### PR TITLE
Making TimeAverageMetricTransformer into a chain of dependency pattern.

### PR DESCRIPTION
**Pull request by michael-mclawhorn**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/protocol/protocols/TimeAverageMetricTransformer.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5260034&bid=17233501)

File: fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/protocol/protocols/TimeAverageMetricTransformer.java [hunk 1]
```diff
@@ -42,17 +42,21 @@
...
      *
      * @param protocolSupportSupplier  A source for the default protocol support used by the makers.
      * @param  makerConfigMap  A collection of maker configurations to use when making time reaggregations.
+     * @param  delegate  A metric transformer to send to if no matching core value is found in this one.ß
      */
     protected TimeAverageMetricTransformer(
             Supplier<ProtocolSupport> protocolSupportSupplier,
-            Map<String, TimeAverageMetricMakerConfig> makerConfigMap
+            Map<String, TimeAverageMetricMakerConfig> makerConfigMap,
+            MetricTransformer delegate
```
Comment (archolewa) [inline fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/protocol/protocols/TimeAverageMetricTransformer.java]: We don't seem to be assigning the delegate to a field or anything...

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/protocol/protocols/TimeAverageMetricTransformer.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5260034&bid=17233501)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/protocol/protocols/TimeAverageMetricTransformer.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5260034&bid=17233501)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/protocol/protocols/TimeAverageMetricTransformer.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5260034&bid=17233501)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/protocol/protocols/TimeAverageMetricTransformer.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5260034&bid=17233501)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/protocol/protocols/TimeAverageMetricTransformer.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5260034&bid=17233501)

File: fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/protocol/protocols/TimeAverageMetricTransformer.java (hunk 1 above)
Comment (michael-mclawhorn) [reply fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/protocol/protocols/TimeAverageMetricTransformer.java:59]: addressed now

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/protocol/protocols/TimeAverageMetricTransformer.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5260034&bid=17233501)

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/protocol/protocols/TimeAverageMetricTransformer.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5260034&bid=17233501)

**Pull request merged**